src/js/form-builder.js

```diff
@@ -98,7 +96,7 @@ const FormBuilder = function(opts, element) {
     }
     let icon = custom.icon || controlClass.icon(type);
     let label = custom.label || controlClass.label(type);
-    let iconClassName = !icon ? custom.iconClassName || `icon-${type}` : '';
+    let iconClassName = !icon ? custom.iconClassName || `icon-${type.replace(/-[\d]{4}$/, '')}` : '';
```

Comment:
Agree on the custom type but since since the custom type key is randomly generated each time, even if there was an icon of say 'autocomplete-2847' those last 4 digits would change on each initialization. The regex removes hyphen and 4 digits which should only be generated by custom fields. If the custom field is actually an extended base type like in the autocomplete example, its possible the autocomplete icon still wants to be used. To make that happen we strip the '-2847' leaving 'icon-autocomplete'. If they don't want the autocomplete icon they can always explicitly define an icon or iconClassName.